runtime/perfetto/db.go

```diff
@@ -65,33 +70,18 @@ type metadataEvent struct {
 	Args map[string]string `json:"args"` // arguments provided for the event
 }
 
-type replicaCacheKey struct {
-	app, version, weaveletId string
-}
-
 // DB is a trace database that stores traces on the local file system.
 //
-// Trace data is stored in a JSON format [1] that is compatible with the
-// Perfetto UI[2].
+// Trace spans are stored in a single sqlite DB table. Each span is stored in
+// a separate row in the table.
 //
-// For each trace span, the following Perfetto events are generated:
-//   - A complete event, encapsulating the span duration.
-//   - A number of metadata events, encapsulating the span events.
-//
-// [1] https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU/preview#
-// [2] https://ui.perfetto.dev/
+// TODO(spetrovic): Separate the database from the Pefetto serving code.
 type DB struct {
-	// Trace data is stored in a sqlite DB spread across three tables:
-	// (1) traces:           trace data in a Perfetto-UI-compattible JSON format
-	// (2) replica_num:      map from colocation group replica id to a replica
-	//                       number
-	// (3) next_replica_num: the next replica number to use for a given
-	//                       colocation group
+	// Trace data is stored in a sqlite DB spread across two tables:
+	// (1) spans:           serialized trace spans
+	// (2) span_attributes: span attributes stored in key,value format
```

Comment:
Good point about encoded_span. I haven't really thought about how table size impacts querying performance.

Btw., attributes need to be separated so that they can be queried (a TODO for now), as they are repeated.

See my earlier comment. I can definitely move encoded_spans into a separate table.

Reply:
Ack